.github/workflows/publish_docker.yml

```diff
@@ -0,0 +1,41 @@
+name: publish docker
+
+on:
+  push:
+    paths:
+      - "src/**"
+      - ".github/workflows/publish_docker.yml"
+    branches:
+      - main
+  pull_request:
+    paths:
+      - "src/**"
+      - ".github/workflows/publish_docker.yml"
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        include:
+          - tag: "7.0" #Debian 11
+            file: Dockerfile
+          - tag: 7.0-jammy # Ubuntu 22.04
+            file: Dockerfile
+          - tag: 7.0-alpine # Alpine 3.17
+            file: Dockerfile.alpine
+          - tag: 7.0-alpine3.16
+            file: Dockerfile.alpine
+    runs-on: ubuntu-latest
```

Comment:
I think we always want to specify our self hosted runners right? Otherwise is uses a github hosted runner?

Reply:
Yea bmx is public repo so we can't even access our self hosted runners

Reply:
I was waiting for a workflow to finish on the master branch last week and took like 30 minutes before figuring out we can't use self hosted anymore 😅

Reply:
Oh yeah I forgot about the public repo thing